x/incentives/keeper/gauge.go

```diff
@@ -350,29 +352,41 @@ func (k Keeper) GetGaugeFromIDs(ctx sdk.Context, gaugeIDs []uint64) ([]types.Gau
 	return gauges, nil
 }
 
-// GetGauges returns upcoming, active, and finished gauges.
+// GetGauges returns upcoming, active, finished, and group gauges.
 func (k Keeper) GetGauges(ctx sdk.Context) []types.Gauge {
-	return k.getGaugesFromIterator(ctx, k.GaugesIterator(ctx))
+	allGauges := k.GetAllGroupsGauges(ctx)
+	allGauges = append(allGauges, k.getGaugesFromIterator(ctx, k.GaugesIterator(ctx))...)
+	sortGaugesByGaugeID(allGauges)
+	return allGauges
 }
 
-// GetNotFinishedGauges returns both upcoming and active gauges.
+// GetNotFinishedGauges returns upcoming, active, and group gauges.
 func (k Keeper) GetNotFinishedGauges(ctx sdk.Context) []types.Gauge {
-	return append(k.GetActiveGauges(ctx), k.GetUpcomingGauges(ctx)...)
+	notFinishedGauges := append(k.GetActiveGauges(ctx), k.GetUpcomingGauges(ctx)...)
+	sortGaugesByGaugeID(notFinishedGauges)
+	return notFinishedGauges
 }
 
-// GetActiveGauges returns active gauges.
+// GetActiveGauges returns active and group gauges.
 func (k Keeper) GetActiveGauges(ctx sdk.Context) []types.Gauge {
-	return k.getGaugesFromIterator(ctx, k.ActiveGaugesIterator(ctx))
+	activeGauges := k.GetAllGroupsGauges(ctx)
+	activeGauges = append(activeGauges, k.getGaugesFromIterator(ctx, k.ActiveGaugesIterator(ctx))...)
+	sortGaugesByGaugeID(activeGauges)
+	return activeGauges
 }
 
 // GetUpcomingGauges returns upcoming gauges.
 func (k Keeper) GetUpcomingGauges(ctx sdk.Context) []types.Gauge {
-	return k.getGaugesFromIterator(ctx, k.UpcomingGaugesIterator(ctx))
+	upcomingGauges := k.getGaugesFromIterator(ctx, k.UpcomingGaugesIterator(ctx))
+	sortGaugesByGaugeID(upcomingGauges)
+	return upcomingGauges
 }
 
 // GetFinishedGauges returns finished gauges.
 func (k Keeper) GetFinishedGauges(ctx sdk.Context) []types.Gauge {
-	return k.getGaugesFromIterator(ctx, k.FinishedGaugesIterator(ctx))
+	finishedGauges := k.getGaugesFromIterator(ctx, k.FinishedGaugesIterator(ctx))
+	sortGaugesByGaugeID(finishedGauges)
+	return finishedGauges
 }
```

Comment:
We should not be modifying these methods. Let's please revert them. They should continue returning regular gauges only as specified by the gauge lifecycle 

The genesis changes should just be:
1. Introduce new method to read group gauges
2. Use it in export genesis
3. In import genesis, bypass ref keys if this is a group gauge